addons/website/static/src/snippets/s_table_of_content/options.js

```diff
@@ -69,23 +78,42 @@ options.registry.TableOfContent = options.Class.extend({
      */
     _generateNav: function (ev) {
         this.options.wysiwyg && this.options.wysiwyg.odooEditor.unbreakableStepUnactive();
-        // We dispose the scrollSpy because the navbar will be updated.
-        this._disposeScrollSpy();
         const $nav = this.$target.find('.s_table_of_content_navbar');
         const $headings = this.$target.find(this.targetedElements);
-        $nav.empty();
-        _.each($headings, el => {
-            const $el = $(el);
-            const id = 'table_of_content_heading_' + _.now() + '_' + _.uniqueId();
-            $('<a>').attr('href', "#" + id)
-                    .addClass('table_of_content_link list-group-item list-group-item-action py-2 border-0 rounded-0')
-                    .text($el.text())
-                    .appendTo($nav);
-            $el.attr('id', id);
-            $el[0].dataset.anchor = 'true';
-        });
-        const exception = (tocEl) => !tocEl.querySelector('.s_table_of_content_navbar a');
-        this._activateScrollSpy(exception);
+        if (this._navbarNeedUpdate(this.oldHeadingsEls, [...$headings])) {
+            // We dispose the scrollSpy because the navbar will be updated.
+            this._disposeScrollSpy();
+            $nav.empty();
+            _.each($headings, el => {
+                const $el = $(el);
+                const id = 'table_of_content_heading_' + _.now() + '_' + _.uniqueId();
+                $('<a>').attr('href', "#" + id)
+                        .addClass('table_of_content_link list-group-item list-group-item-action py-2 border-0 rounded-0')
+                        .text($el.text())
+                        .appendTo($nav);
+                $el.attr('id', id);
+                $el[0].dataset.anchor = 'true';
+            });
+            const exception = (tocEl) => !tocEl.querySelector('.s_table_of_content_navbar a');
+            this._activateScrollSpy(exception);
+            this.oldHeadingsEls = [...($headings.clone())];
+        }
+    },
+
+    /**
+    * The navbar needs to be updated if its content before the change of the
+    * DOM is different fom the one after the change of the DOM.
+    */
+    _navbarNeedUpdate(oldHeadingsEls, newHeadingsEls) {
```

Comment:
- Wrong indentation of the JS DOC. Try to install/fix something on your editor so it is correctly indented automatically :+1: 
- I think it deserves a bit more explanation why this is necessary (either here or the caller). IMO you should not make a function out of this you only call it once and it is very small. A computation of a variable `areHeadingsEqual` and a commented `if (!areHeadingsEqual)` would be way clearer IMO.
- If we would keep this function: `_navbarNeedUpdate` -> `_navbarNeedsUpdate` (but let's not keep it)